frontend/src/components/QueryCodeEditor/index.tsx

```diff
@@ -10,10 +10,16 @@ import 'codemirror/addon/hint/show-hint';
 import 'codemirror/addon/hint/sql-hint';
 import 'codemirror/addon/hint/javascript-hint';
 import 'codemirror/addon/hint/show-hint.css';
+import { setNewCodeRoutine } from '../../scenes/constructor/routines';
+import { useDispatch, useSelector } from 'react-redux';
+import { IAppState } from '../../common/models/store/IAppState';
 
 const QueryEditor: React.FC<ITables> = ({ tables }) => {
-  const [query, setQuery] = useState('');
-
+  const query = useSelector((state: IAppState) => state.app.qur);
+  const dispatch = useDispatch();
+  function changeCode(e: string) {
+    dispatch(setNewCodeRoutine.trigger({ code: e }));
+  }
```

Comment:
why not an arrow function? Also, there shouldn't be any `dispatch` in the `QueryCodeEditor` component, it shouldn't handle any app's logic. Instead, it should accept this function in props from the parent component